app/src/test/java/org/mozilla/fenix/home/blocklist/BlocklistMiddlewareTest.kt

```diff
@@ -341,4 +350,55 @@ class BlocklistMiddlewareTest {
             (store.state.recentSyncedTabState as RecentSyncedTabState.Success).tabs.single()
         )
     }
+
+    @Test
+    fun `WHEN the recent synced tab state is changed to None or Loading THEN the middleware does not change the state`() {
+        val blockedHost = "https://www.mozilla.org"
+        every { mockSettings.homescreenBlocklist } returns setOf(blockedHost.stripAndHash())
+        val middleware = BlocklistMiddleware(blocklistHandler)
+        val store = AppStore(
+            AppState(),
+            middlewares = listOf(middleware)
+        )
+
+        store.dispatch(
+            AppAction.RecentSyncedTabStateChange(
+                RecentSyncedTabState.None
+            )
+        ).joinBlocking()
+
+        assertEquals(RecentSyncedTabState.None, store.state.recentSyncedTabState)
+    }
+
+    @Test
+    fun `WHEN all recently synced submitted tabs are blocked THEN the recent synced tab state should be set to None`() {
+        val blockedHost = "https://www.mozilla.org"
+        val blockedTab = RecentSyncedTab(
+            deviceDisplayName = "",
+            deviceType = mock(),
+            title = "",
+            url = "https://www.mozilla.org",
+            previewImageUrl = null
+        )
+
+        every { mockSettings.homescreenBlocklist } returns setOf(blockedHost.stripAndHash())
+        val middleware = BlocklistMiddleware(blocklistHandler)
+        val store = AppStore(
+            AppState(),
+            middlewares = listOf(middleware)
+        )
+
+        store.dispatch(
+            AppAction.RecentSyncedTabStateChange(
+                RecentSyncedTabState.Success(
+                    listOf(blockedTab)
+                )
+            )
+        ).joinBlocking()
+
+        assertEquals(
+            RecentSyncedTabState.None,
+            store.state.recentSyncedTabState
+        )
+    }
```

Comment:
These look good! I would prefer to also include a test that captures the behavior of switching to the next recent synced tab when one is blocked, as well.